File: android/app/src/main/java/dk/cipher/remotedisplay/models/Display.kt
package dk.cipher.remotedisplay.models

data class Display(val lines: Array<Line> = Array(Specs.lineCount) { Line() }) {
    object Specs {
        const val lineCount = 2
        const val charCount = lineCount * Line.Specs.charCount
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Display) return false
        return lines.contentEquals(other.lines)
    }

    override fun hashCode(): Int {
        return lines.contentHashCode()
    }
}

File: apple/RemoteDisplay/UI/Models/Display.swift
//
//  Display.swift
//  RemoteDisplay
//
//  Created by Júlio Flores on 22/08/2026.
//

struct Display {
	enum Specs {
		static let lineCount: Int = 2
		static var charCount: Int {
			lineCount * Line.Specs.charCount
		}
	}

	@Fixed var lines = [Line](repeating: .init(), count: Specs.lineCount)
}
